### PR TITLE
Implemenet New Interop Test: Cpp Unimplemented Service

### PR DESCRIPTION
**Pull request by ncteisen**
Wrote code for unimplemented_service, which calls UnimplementedService/UnimplementedCall, then asserts a Status::UNIMPLEMENTED was returned.

Also changed run_interop_tests.py to call the new test. I ran the full tests suite locally and everything passed.

### CODE REVIEW

File: test/cpp/interop/interop_client.cc [hunk 1]
```diff
@@ -1002,6 +1007,27 @@ bool InteropClient::DoCustomMetadata() {
   return true;
 }
 
+bool InteropClient::DoUnimplementedService() {
+  gpr_log(GPR_DEBUG, "Sending a request for an unimplemented service...");
+
+  Empty request = Empty::default_instance();
+  Empty response = Empty::default_instance();
```
Comment (ncteisen) [inline test/cpp/interop/interop_client.cc]: Done

**Comment by ncteisen** [comment]
Changed Empty::default_instance() to just use the default ctor and everything still looks good.

**Comment by ncteisen** [comment]
Jenkins failure seems to be due to flakiness. The test suite runs fine on my computer

**Comment by yang-g** [comment]
Clang-format?
```
--- /local-code/test/cpp/interop/client.cc	2016-10-28 20:55:54.637759787 +0000
+++ /tmp/tmp.fmZKlH1OTK	2016-10-28 21:07:24.457425919 +0000
@@ -54,34 +54,35 @@
 DEFINE_string(server_host, "127.0.0.1", "Server host to connect to");
 DEFINE_string(server_host_override, "foo.test.google.fr",
               "Override the server host which is sent in HTTP header");
-DEFINE_string(test_case, "large_unary",
-              "Configure different test cases. Valid options are:\n\n"
-              "all : all test cases;\n"
-              "cancel_after_begin : cancel stream after starting it;\n"
-              "cancel_after_first_response: cancel on first response;\n"
-              "client_compressed_streaming : compressed request streaming with "
-              "client_compressed_unary : single compressed request;\n"
-              "client_streaming : request streaming with single response;\n"
-              "compute_engine_creds: large_unary with compute engine auth;\n"
-              "custom_metadata: server will echo custom metadata;\n"
-              "empty_stream : bi-di stream with no request/response;\n"
-              "empty_unary : empty (zero bytes) request and response;\n"
-              "half_duplex : half-duplex streaming;\n"
-              "jwt_token_creds: large_unary with JWT token auth;\n"
-              "large_unary : single request and (large) response;\n"
-              "oauth2_auth_token: raw oauth2 access token auth;\n"
-              "per_rpc_creds: raw oauth2 access token on a single rpc;\n"
-              "ping_pong : full-duplex streaming;\n"
-              "response streaming;\n"
-              "server_compressed_streaming : single request with compressed "
-              "server_compressed_unary : single compressed response;\n"
-              "server_streaming : single request with response streaming;\n"
-              "slow_consumer : single request with response streaming with "
-              "slow client consumer;\n"
-              "status_code_and_message: verify status code & message;\n"
-              "timeout_on_sleeping_server: deadline exceeds on stream;\n"
-              "unimplemented_method: client calls an unimplemented method;\n"
-              "unimplemented_service: client calls an unimplemented service;\n");
+DEFINE_string(
+    test_case, "large_unary",
+    "Configure different test cases. Valid options are:\n\n"
+    "all : all test cases;\n"
+    "cancel_after_begin : cancel stream after starting it;\n"
+    "cancel_after_first_response: cancel on first response;\n"
+    "client_compressed_streaming : compressed request streaming with "
+    "client_compressed_unary : single compressed request;\n"
+    "client_streaming : request streaming with single response;\n"
+    "compute_engine_creds: large_unary with compute engine auth;\n"
+    "custom_metadata: server will echo custom metadata;\n"
+    "empty_stream : bi-di stream with no request/response;\n"
+    "empty_unary : empty (zero bytes) request and response;\n"
+    "half_duplex : half-duplex streaming;\n"
+    "jwt_token_creds: large_unary with JWT token auth;\n"
+    "large_unary : single request and (large) response;\n"
+    "oauth2_auth_token: raw oauth2 access token auth;\n"
+    "per_rpc_creds: raw oauth2 access token on a single rpc;\n"
+    "ping_pong : full-duplex streaming;\n"
+    "response streaming;\n"
+    "server_compressed_streaming : single request with compressed "
+    "server_compressed_unary : single compressed response;\n"
+    "server_streaming : single request with response streaming;\n"
+    "slow_consumer : single request with response streaming with "
+    "slow client consumer;\n"
+    "status_code_and_message: verify status code & message;\n"
+    "timeout_on_sleeping_server: deadline exceeds on stream;\n"
+    "unimplemented_method: client calls an unimplemented method;\n"
+    "unimplemented_service: client calls an unimplemented service;\n");
 DEFINE_string(default_service_account, "",
               "Email of GCE default service account");
 DEFINE_string(service_account_key_file, "",
--- /local-code/test/cpp/interop/interop_client.cc	2016-10-28 20:55:54.637759787 +0000
+++ /tmp/tmp.V2HZ5QN2hm	2016-10-28 21:07:24.625425838 +0000
@@ -1014,11 +1014,9 @@
   Empty response;
   ClientContext context;
 
-  UnimplementedService::Stub* stub =
-      serviceStub_.GetUnimplementedServiceStub();
+  UnimplementedService::Stub* stub = serviceStub_.GetUnimplementedServiceStub();
 
-  Status s =
-      stub->UnimplementedCall(&context, request, &response);
+  Status s = stub->UnimplementedCall(&context, request, &response);
 
   if (!AssertStatusCode(s, StatusCode::UNIMPLEMENTED)) {
     return false;

FAILED: tools/distrib/clang_format_code.sh [ret=1, pid=17172]
```

**Pull request merged**